runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -851,27 +853,53 @@ TR_J9InlinerPolicy::genCodeForUnsafeGetPut(TR::Node* unsafeAddress,
                                        TR::TreeTop* indirectAccessTreeTop,
                                        bool needNullCheck, bool isUnsafeGet,
                                        bool conversionNeeded,
+                                       bool arrayCheckNeeded, bool arrayBlockNeeded,
+                                       bool checksNeeded,
                                        TR::Node* orderedCallNode = NULL)
    {
+   // There are 4 possible cases determining which checks and blocks should be added to the generated IL trees:
+   // 1.) checksNeeded && arrayCheckNeeded && arrayBlockNeeded
+   //     (i.e.: (offheap AND object type is unknown at compile time) OR conversionNeeded)
+   //     - checks: NULLCHK, ArrayCHK, LowtagCHK
+   //     - blocks: arrayDirectAccess, directAccess, indirectAccess
+   // 2.) checksNeeded && !arrayCheckNeeded && !arrayBlockNeeded
+   //     (i.e.: gencon AND object type is unknown at compile time AND !conversionNeeded)
```

Comment:
It doesn't look like `conversionNeeded` is used any longer after the new `arrayCheckNeeded`, `arrayBlockNeeded` and `checksNeeded` are introduced.  If that's correct, the parameter should be removed and the comments adjusted.